dev/doc/build-system.dune.md

```diff
@@ -112,13 +112,24 @@ ml files in quick mode.
 Dune also provides targets for documentation, testing, and release
 builds, please see below.
 
-## Documentation and testing targets
+## Testing and documentation targets
 
-Coq's test-suite can be run with `dune runtest`; given that `dune`
-still invokes the test-suite makefile, the environment variable
-`NJOBS` will control the value of the `-j` option that is passed to
-make; common call `NJOBS=8 dune runtest`. This will be resolved in the
-future once the test suite is ported to Dune rules.
+There are two ways to run the test suite using Dune:
+
+- After building Coq with `make -f Makefile.dune world`, you can run the test-suite
+  in place, generating output files in the source tree
+  by running `make -C test-suite` from the top directory of the source tree
+  (equivalent to running `make test-suite` from the `test-suite` directory).
+  This permits incremental usage since output files will be preserved.
+
+- You can also run the test suite in a hygienic way using `make -f
+  Makefile.dune test-suite` or `dune runtest`. This is convenient for
+  full runs from scratch, for instance in CI.
+
+Since `dune` still invokes the test-suite makefile, the
+environment variable `NJOBS` is used to set the `-j` option
+that is passed to make (for example, with the command `NJOBS=8 dune runtest`). This use of `NJOBS`
+will be removed when the test suite is fully ported to Dune.
```

Comment:
```suggestion
  Since `dune` still invokes the test-suite makefile, the
  environment variable `NJOBS` is used to set the `-j` option
  that is passed to make (for example, with the command
  `NJOBS=8 dune runtest`). This use of `NJOBS` will be
  removed when the test suite is fully ported to Dune.
```